lusid-sdk-python/tests/test_transactions.py

```diff
@@ -0,0 +1,233 @@
+                                    settlement_date=effective_date,
+                                    units=100.0,
+                                    transaction_price=models.TransactionPrice(12.3),
+                                    total_consideration=models.CurrencyAndAmount(100 * 12.3, "GBP"),
+                                    source="Client")
+        # add the trade
+        upsert_response = self.client.upsert_transactions(scope, portfolio_id, [transaction_request])
+
+        # get the trade again
+        returned_transaction = self.client.get_transactions(scope, portfolio_id )
+        assert returned_transaction.count == 1
+        assert transaction_request.transaction_id == returned_transaction.values[0].transaction_id
+
+    def test_zzload_cash_transactions(self):
+
+        uuid_gen = uuid.uuid4()
+        scope = "finbourne"
+        effective_date = datetime(2018, 1, 1, tzinfo=pytz.utc)
+
+        # portfolio request
+        original_portfolio_id = "Id-" + str(uuid_gen)
+        request = models.CreateTransactionPortfolioRequest("Portfolio-" + str(uuid_gen),
+                                                           original_portfolio_id,
+                                                           base_currency="GBP",
+                                                           created=effective_date)
+        # create the portfolio
+        portfolio_response = self.client.create_portfolio(scope, request)
+
+        self.assertEqual(portfolio_response.id.code, request.code)
+
+        portfolio_id = portfolio_response.id.code
+
+        # create the transaction
+        # Cash instruments are identified using CCY_ followed by the ISO currency codes.
+        # Cash instruments do not need to be created before use
+        transaction_request = models.TransactionRequest(
+                                    transaction_id=str(uuid.uuid4()),
+                                    type="FundsIn",
+                                    instrument_uid="CCY_GBP",
+                                    transaction_date=effective_date,
+                                    settlement_date=effective_date,
+                                    units=100.0,
+                                    transaction_price=models.TransactionPrice(0.0),    # this is not included in java version
+                                    total_consideration=models.CurrencyAndAmount(0.0, "GBP"),
+                                    source="Client")
+
+        # add the trade
+        upsert_response = self.client.upsert_transactions(scope, portfolio_id, [transaction_request])
+
+        # get the trade again
+        returned_transaction = self.client.get_transactions(scope, portfolio_id)
+        assert returned_transaction.count == 1
+        assert transaction_request.transaction_id == returned_transaction.values[0].transaction_id
+
+    def test_load_otc_instrument_transaction(self):
+        uuid_gen = uuid.uuid4()
```

Comment:
It looks like you repeat these same steps for all the tests, it might be worth creating your portfolio at the start where you load your instruments as well as there is no need for multiple portfolios here you can have all transactions in the same portfolio. 

When you call get_transaction at the end to check it is correct you can use a filter to filter to the transaction you want by its transaction id.

It would look like this - 

filter="transactionId eq '{}'".format(transaction_id)

Reply:
i did suggest a little more reuse, but riz wanted to keep things as they were to be nice and clear for the customers